numpy/lib/financial.py

```diff
@@ -645,7 +645,7 @@ def rate(nper, pmt, pv, fv, when='end', guess=None, tol=None, maxiter=100):
         rn = rnp1
     if not close:
         # Return nan's in array of the same shape as rn
-        return np.nan + rn
+        return default_type(np.nan)+rn
```

Comment:
Also, please add spaces around the operator here, as per PEP 8, that is typically needed (there are some exceptions, this is not one of them though IMO).

Reply:
Thanks @seberg for review. I will insert whitespace as per PEP 8 style guidelines. Will also write one test for testing out the new change.